go/store/nbs/conjoiner.go

```diff
@@ -197,6 +197,13 @@ func conjoinTables(ctx context.Context, conjoinees []tableSpec, p tablePersister
 			return
 		})
 	}
+	defer func() {
```

Comment:
This was the primary leak, AFAICT. For a conjoin, all table files are newly opened and read, instead of using our existing table file indexes. The newly conjoined file is also opened as a chunk source down below. All of these chunk sources were previously leaked.